From f31c1c179b32535ed36476b8935d3ae3964a6819 Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Tue, 5 Mar 2019 10:12:35 -0800
Subject: [PATCH 0536/1851] drm: xlnx: pl_disp: handle software vblank

commit f9c7d196b913e584adb5371f925e502dab77fa40 from
https://github.com/Xilinx/linux-xlnx.git

During crtc begin/disable, software vblank should be on/off
accordingly. Without this functionality, 4.19 kernel gives
warning message for the initial run.

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_pl_disp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
index 55f7522873d2..e0e68c5f6826 100644
--- a/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
+++ b/drivers/gpu/drm/xlnx/xlnx_pl_disp.c
@@ -308,6 +308,7 @@ static inline struct xlnx_pl_disp *drm_crtc_to_dma(struct drm_crtc *crtc)
 static void xlnx_pl_disp_crtc_atomic_begin(struct drm_crtc *crtc,
 					   struct drm_crtc_state *old_state)
 {
+	drm_crtc_vblank_on(crtc);
 	spin_lock_irq(&crtc->dev->event_lock);
 	if (crtc->state->event) {
 		/* Consume the flip_done event from atomic helper */
@@ -359,6 +360,7 @@ static void xlnx_pl_disp_crtc_atomic_disable(struct drm_crtc *crtc,
 
 	xlnx_pl_disp_plane_disable(crtc->primary);
 	xlnx_pl_disp_clear_event(crtc);
+	drm_crtc_vblank_off(crtc);
 	xlnx_bridge_disable(xlnx_pl_disp->vtc_bridge);
 }
 
-- 
2.31.1

